src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleExtensions.scala

```diff
@@ -253,6 +257,18 @@ class GoogleExtensions(val directoryDAO: DirectoryDAO, val accessPolicyDAO: Acce
     } yield key
   }
 
+  def getPetServiceAccountToken(user: WorkbenchUser, project: GoogleProject): Future[String] = {
+    getPetServiceAccountKey(user, project).flatMap { key =>
+      getAccessTokenUsingJson(key)
+    }
+  }
+
+  def getAccessTokenUsingJson(saKey: String) : Future[String] = Future {
+    val keyStream = new ByteArrayInputStream(saKey.getBytes)
+    val credential = ServiceAccountCredentials.fromStream(keyStream).createScoped(Seq(PlusScopes.USERINFO_EMAIL, PlusScopes.USERINFO_PROFILE, StorageScopes.DEVSTORAGE_READ_ONLY).asJava)
```

Comment:
I'm curious if `DEVSTORAGE_READ_ONLY` scopes are really needed here? Would it make sense to use the same scopes for the pet token as the scopes of the requesting user?

Reply:
using the same scopes as the requesting user would break saturn UI. they're not allowed to request storage scopes for users, which is why they need to use pets to get around that restriction.

i like the idea of specifying the scopes. since it seems like there are several different use-cases with different scopes requirements.